costs/es_request_constructor.go

```diff
@@ -39,133 +45,143 @@ var paramNameToFuncPtr = map[string]func([]string) []paramAggrAndName{
-	res := make([]paramAggrAndName, 1)
-	res[0] = paramAggrAndName{
-		paramName: "week",
-		paramAggr: elastic.NewDateHistogramAggregation().
-			Field("usage_start_date").Interval("week")}
-	return res
+	return []paramAggrAndName{
+		paramAggrAndName{
+			name: "week",
+			aggr: elastic.NewDateHistogramAggregation().
+				Field("usage_start_date").Interval("week"),
+		},
+	}
 }
 
-// createAggregationPerMonth : Creates and returns a new []paramAggrAndName of size 1 which creates a
+// createAggregationPerMonth creates and returns a new []paramAggrAndName of size 1 which creates a
 // date histogram aggregation on the field 'usage_start_date' with a time range of a month
 func createAggregationPerMonth(_ []string) []paramAggrAndName {
-	res := make([]paramAggrAndName, 1)
-	res[0] = paramAggrAndName{
-		paramName: "month",
-		paramAggr: elastic.NewDateHistogramAggregation().
-			Field("usage_start_date").Interval("month")}
-	return res
+	return []paramAggrAndName{
+		paramAggrAndName{
+			name: "month",
+			aggr: elastic.NewDateHistogramAggregation().
+				Field("usage_start_date").Interval("month"),
+		},
+	}
 }
 
-// createAggregationPerYear : Creates and returns a new []paramAggrAndName of size 1 which creates a
+// createAggregationPerYear creates and returns a new []paramAggrAndName of size 1 which creates a
 // date histogram aggregation on the field 'usage_start_date' with a time range of a year
 func createAggregationPerYear(_ []string) []paramAggrAndName {
-	res := make([]paramAggrAndName, 1)
-	res[0] = paramAggrAndName{
-		paramName: "year",
-		paramAggr: elastic.NewDateHistogramAggregation().
-			Field("usage_start_date").Interval("year")}
-	return res
+	return []paramAggrAndName{
+		paramAggrAndName{
+			name: "year",
+			aggr: elastic.NewDateHistogramAggregation().
+				Field("usage_start_date").Interval("year"),
+		},
+	}
 }
 
-// createAggregationPerTag : Creates and return a new []paramAggrAndName of size 2, which creates two
-// agregation. The first aggregation is a FilterAggregation on the field 'tag.key', and with a value of
+// createAggregationPerTag creates and returns a new []paramAggrAndName of size 2 which consits
+// of two aggregations that are required for the tag param.
+// The first aggregation is a FilterAggregation on the field 'tag.key', and with a value of
 // the tag key passed in the parameter 'paramSplit' in the form "user:<TAG_KEY_VALUE>".
 // The second aggregation is a TermsAggregation that creates bucket aggregation on the field
 // 'tag.value'.
 // No SubAggregation is created in this fuction, as it needs to be created in the nestAggregation function
 func createAggregationPerTag(paramSplit []string) []paramAggrAndName {
 	res := make([]paramAggrAndName, 2)
 	res[0] = paramAggrAndName{
-		paramName: "tag_key",
-		paramAggr: elastic.NewFilterAggregation().
+		name: "tag_key",
+		aggr: elastic.NewFilterAggregation().
 			Filter(elastic.NewTermQuery("tag.key", fmt.Sprintf("user:%v", paramSplit[1])))}
```

Comment:
Why don't you do exactly the same thing as above in your code ? I mean, you can return `[]paramAggrAndName{}` and put two `paramAggrAndName` inside.

Reply:
You're right, I didn't catch that, I will change it to be the same as the others `aggregationBuilder` functions